### PR TITLE
Fedora fixes

### PR DESCRIPTION
**Pull request by bkmgit**
Addressing https://github.com/HVML/PurC/issues/40

### CODE REVIEW

**Comment by VincentWei** [comment]
Many thanks for your pull requests! It looks very good. We will review it asap.

**Review by yongkangl — APPROVED** [review]

**Comment by yongkangl** [comment]
Oops, confirmed the first one but looks all patches are approved, is it possible to cancel the action?

File: Source/test/dvobjs/test_dvobjs_system.cpp [hunk 1]
```diff
@@ -880,19 +880,18 @@ purc_variant_t system_timezone(purc_variant_t dvobj, const char* name)
...
-                path[nr_bytes] = 0;
-                timezone = path + sizeof(PURC_SYS_TZ_DIR) - 1;
-            }
+            nr_bytes = readlink(PURC_SYS_TZ_FILE, path, sizeof(path)-1);
+            if (nr_bytes > 0 ) path[nr_bytes] = '\0';
+            if ((nr_bytes > 0 ) &&
+                (strstr(path, PURC_SYS_TZ_DIR) != NULL))
+                     timezone = strstr(path,PURC_SYS_TZ_DIR) +
+                         sizeof(PURC_SYS_TZ_DIR) - 1;
         }
```
Comment (yongkangl) [inline Source/test/dvobjs/test_dvobjs_system.cpp:896]: Assert here for detecting no valid timezone is retrieved?

**Comment by LTreeshu** [comment]
good job!

File: Source/test/dvobjs/test_dvobjs_system.cpp (hunk 1 above)
Comment (bkmgit) [reply Source/test/dvobjs/test_dvobjs_system.cpp:901]: @yongkangl Thanks for the review. Modified to log an error and return if no valid timezone is retrieved.

**Review by yongkangl — APPROVED** [review]
Thanks. Looks good.

**Comment by VincentWei** [comment]
I revised `pcdvobjs_get_current_timezone()` for the correct coding style:

```c
bool
pcdvobjs_get_current_timezone(char *buff, size_t sz_buff)
{
    const char *timezone;
    char path[PATH_MAX + 1];

    const char* env_tz = getenv("TZ");
    if (env_tz) {
        if (env_tz[0] == ':')
            timezone = env_tz + 1;
        if (!pcdvobjs_is_valid_timezone(timezone)) {
            timezone = "posixrules";
        }
    }
    else {
        ssize_t nr_bytes;
        nr_bytes = readlink(PURC_SYS_TZ_FILE, path, sizeof(path)-1);
        if (nr_bytes > 0)
             path[nr_bytes] = '\0';
        if ((nr_bytes > 0) &&
            (strstr(path, PURC_SYS_TZ_DIR) != NULL)) {
                    timezone = strstr(path, PURC_SYS_TZ_DIR) +
                                    sizeof(PURC_SYS_TZ_DIR) - 1;
        }
        else {
            purc_set_error(PURC_ERROR_BAD_SYSTEM_CALL);
            purc_log_error("Cannot determine timezone.\n");
            goto failed;
        }
    }

    if (strlen(timezone) >= sz_buff) {
        purc_set_error(PURC_ERROR_TOO_SMALL_BUFF);
        goto failed;
    }

    strcpy(buff, timezone);
    return true;

failed:
    return false;
}
```

**Comment by VincentWei** [comment]
I have merged the commits. However, I will revise the code for the correct coding style.

Thank @bkmgit and @yongkangl  again for your efforts!

**Comment by bkmgit** [comment]
谢谢  . Would it be worth using tools such as [indent](https://www.gnu.org/software/indent/) or [astyle](https://astyle.sourceforge.net/)?

**Comment by VincentWei** [comment]
> 谢谢 . Would it be worth using tools such as [indent](https://www.gnu.org/software/indent/) or [astyle](https://astyle.sourceforge.net/)?

The coding style of PurC is basically like the one of Linux kernel:

<https://www.kernel.org/doc/html/v4.10/process/coding-style.html>

I think the programmer should write code in right coding style instead of relying on a cleaning tool. We only use the tools when we want to copy or import some old code into a new project.

**Comment by bkmgit** [comment]
老师谢谢。 本周将更新。

**Comment by VincentWei** [comment]
> 老师谢谢。 本周将更新。

I have merged your changes and revised the code for the correct coding style.

I will push the commits to this repository in a few days.

**Pull request merged**